Matchismo/Matchismo/Model/SetCardDeck.mm

```diff
@@ -0,0 +1,36 @@
+//
+//  SetCardDeck.m
+//  Matchismo
+//
+//  Created by Otto Olkkonen on 15/06/2022.
+//
+
+#import "SetCardDeck.h"
+
+#import "SetCard.h"
+
+@implementation SetCardDeck
+
+- (instancetype)initWithShapes:(NSArray *)shapes Colors:(NSArray *)colors {
+  if (self = [super init]) {
+    if (shapes.count != 3 || colors.count != 3) return nil;
```

Comment:
Objective-C doesn't have a direct equivalent of Swift's failable initialisers, so returning `nil` like this from an init method isn't the right approach.
Firstly it'll give you a memory leak (as the caller has already invoked `alloc`).  But also it could also cause really unexpected behaviour as someone calling `auto myVar = [[SetCardDeck alloc] initWithShapes:shape colors:colors]` will never imagine that the result could be `nil`. 
If it's a _really_ bad error then you could throw an exception or something (and hence crash the app), but in most cases it would probably be better to use a factory method instead.  So something like:
```
+ (nullable instancetype)setCardDeckWithShapes:(NSArray *)shapes Colors:(NSArray *)colors {
  if (shapes.count != 3 || colors.count != 3) {
     return nil;
  }
  return [[self alloc] initWithShapes:shapes colors:colors];
}
```